contrib/hyperscan/matching/input_matchers/source/matcher.cc

```diff
@@ -95,12 +59,88 @@ bool Matcher::match(absl::optional<absl::string_view> input) {
       },
       &matched);
   if (err != HS_SUCCESS && err != HS_SCAN_TERMINATED) {
-    ENVOY_LOG_MISC(error, "unable to scan, error code {}.", err);
+    ENVOY_LOG_MISC(error, "unable to scan, error code {}", err);
   }
 
   return matched;
 }
 
+std::string Matcher::replaceAll(absl::string_view value, absl::string_view substitution) const {
+  // Find matched pair.
+  using Found = std::pair<unsigned long long, unsigned long long>;
```

Comment:
WDYT of naming this something like Bounds, and commenting whether .second is inclusive or not?

Or is it a start-position and length? Or something else?

I'm wondering why that has to be 'unsigned long long'? Usually we don't spell that out in Envoy and would use uint64_t.

Reply:
It is a good idea. I am not good at naming, and "bounds" sounds great.

I will leave some comments here.

It is just because Hyperscan uses `unsigned long long`, I can just change them to `uint64_t`.